envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotUpdater.kt

```diff
@@ -117,14 +122,20 @@ class SnapshotUpdater(
                 if (properties.enabledCommunicationModes.ads) {
                     lastAdsSnapshot = snapshotFactory.newSnapshot(states, clusters, ADS)
                 }
-
                 val updateResult = UpdateResult(
                     action = Action.ALL_SERVICES_GROUP_ADDED,
                     adsSnapshot = lastAdsSnapshot,
                     xdsSnapshot = lastXdsSnapshot
                 )
                 globalSnapshot = updateResult
                 updateResult
+            }.scan { previous: UpdateResult?, actual: UpdateResult? ->
+                val emptyUpdateResult = emptyUpdateResult()
+                snapshotChangeAuditor.audit(previous ?: emptyUpdateResult, actual ?: emptyUpdateResult)
+                    .publishOn(globalSnapshotAuditScheduler)
+                    .subscribeOn(globalSnapshotAuditScheduler)
+                    .subscribe().apply { auditDisposables.add(this) }
```

Comment:
why we need to use disposable here?

Reply:
imho subscribeon and subscribe will be enough u do not need to do close or publishOn here